sql/src/test/java/io/crate/planner/fetch/FetchPushDownTest.java

```diff
@@ -0,0 +1,128 @@
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.planner.fetch;
+
+import com.google.common.collect.ImmutableList;
+import io.crate.analyze.OrderBy;
+import io.crate.analyze.QuerySpec;
+import io.crate.analyze.WhereClause;
+import io.crate.metadata.ReferenceIdent;
+import io.crate.metadata.ReferenceInfo;
+import io.crate.metadata.TableIdent;
+import io.crate.planner.RowGranularity;
+import io.crate.planner.symbol.Reference;
+import io.crate.planner.symbol.Symbol;
+import io.crate.types.DataTypes;
+import org.junit.Test;
+
+import static io.crate.testing.TestingHelpers.isReference;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+
+public class FetchPushDownTest {
+
+    private static final TableIdent TABLE_IDENT = new TableIdent("s", "t");
+    private static final Reference REF_I = new Reference(
+            new ReferenceInfo(
+                    new ReferenceIdent(new TableIdent("s", "t"), "i"),
+                    RowGranularity.DOC,
+                    DataTypes.INTEGER
+            )
+    );
+
+    private static final Reference REF_A = new Reference(
+            new ReferenceInfo(
+                    new ReferenceIdent(new TableIdent("s", "t"), "a"),
+                    RowGranularity.DOC,
+                    DataTypes.STRING
+            )
+    );
+
+    @Test
+    public void testLimitIsPushedDown() throws Exception {
+        QuerySpec qs = new QuerySpec();
+        qs.outputs(ImmutableList.<Symbol>of(REF_I, REF_A));
+        qs.limit(10);
+        qs.offset(100);
+
+        QuerySpec sub = FetchPushDown.pushDown(qs, TABLE_IDENT);
+        assertThat(sub.limit(), is(110));
+        assertThat(sub.offset(), is(0));
+
+        assertThat(qs.limit(), is(10));
+        assertThat(qs.offset(), is(100));
+    }
+
+    @Test
+    public void testWhereIsPushedDown() throws Exception {
+        QuerySpec qs = new QuerySpec();
+        qs.outputs(ImmutableList.<Symbol>of(REF_I, REF_A));
+        qs.where(WhereClause.NO_MATCH);
+
+        QuerySpec sub = FetchPushDown.pushDown(qs, TABLE_IDENT);
+
+        assertThat(sub.where(), is(WhereClause.NO_MATCH));
+        assertThat(qs.where(), is(WhereClause.MATCH_ALL));
+    }
+
+
```

Comment:
please add a test with a more complex where clause, e.g.: something like:

where t.i < 8 and t.a > 3 and (t.i + t.a) < 10